### ISSUE TITLE
Nibabel throws error when saving images using the `int64` datatype due to compatibility issues

### ISSUE THREAD

**Issue by joshuacwnewton** [spinalcordtoolbox/spinalcordtoolbox#4408]
Failing CI run: https://github.com/spinalcordtoolbox/spinalcordtoolbox/actions/runs/8341947827

Full error traceback:

```python
File "/home/runner/sct_6.3.dev0/testing/api/test_deepseg_sc.py", line 85, in test_uncrop_image
    nii = nib.nifti1.Nifti1Image(data_in, affine)
  File "/home/runner/sct_6.3.dev0/python/envs/venv_sct/lib/python3.9/site-packages/nibabel/nifti1.py", line 1836, in __init__
    alert_future_error(
  File "/home/runner/sct_6.3.dev0/python/envs/venv_sct/lib/python3.9/site-packages/nibabel/deprecated.py", line 120, in alert_future_error
    raise error_class(f'{msg} {error_rec}'.strip())
ValueError: """Image data has type int64, which may cause incompatibilities with other tools. 
               To use this type, pass an explicit header or dtype argument to Nifti1Image()."""
```

Upstream context:

- https://github.com/nipy/nibabel/issues/1089

(The reason that this error is now happening on `master` is due to an upgrade to the `ivadomed` package (2.9.10), which enables `nibabel` to be upgraded, which happened automatically on master once I created the new ivadomed release.)

I encountered this exact error on https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/4332, too. (See: https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/4332/commits/cb7157ff4c202dd87b3ce59efdd4826897bdd413) So, the errors will be fixed when #4332 is merged.

However, given how "general" this error message is, I am concerned that there will be instances of this issue elsewhere in our code that aren't caught by the test suite. So, I'd like to consider adding a fix to the `Image` constructor (either by passing `dtype=np.int64` when the input array is of type `np.int64` to preserve the old behavior, or by converting input arrays to `np.int32` before loading them to fix the potential incompatibilities hinted at by the code).

**Comment by joshuacwnewton** [comment]
From my investigation in #4332, since this error only occurs when `nib.Nifti1Image` is called without the `header` or `dtype` arguments, we don't have to worry about this within `Image`. 

Instead, we only have to worry about this for calls to `nib.Nifti1Image` without the `header` argument. Meaning, this issue will probably be solved when we address https://github.com/spinalcordtoolbox/spinalcordtoolbox/issues/3232#issuecomment-1917645212, since it's meant to replace all instances of `nib.Nifti1Image` + `nib.save()` with `Image` and `Image.save()`. And, that issue was on my TODO list anyway.

So, I think we can let this issue be solved by #4332 + #3232? Unless we want to consider the more long-term fix of not producing _any_ `int64` images from SCT. But, I would have to spend some time understanding what these "int64 compatibility" issues are, and whether we want to mitigate them.

**Comment by joshuacwnewton** [comment]
The only other thing we might want to consider is the more long-term fix of not producing _any_ `int64` images from SCT. But, I would have to spend some time understanding what these "int64 compatibility" issues are, and whether we want to mitigate them.
